src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php

```diff
@@ -5,25 +5,73 @@
 use Backend\Core\Engine\Authentication as BackendAuthentication;
 use Backend\Core\Engine\Base\AjaxAction as BackendBaseAJAXAction;
 use Backend\Core\Language\Language;
+use Backend\Modules\MediaLibrary\Component\StorageProvider\LocalStorageProvider;
 use Backend\Modules\MediaLibrary\Domain\MediaFolder\Exception\MediaFolderNotFound;
+use Backend\Modules\MediaLibrary\Domain\MediaFolder\MediaFolderRepository;
 use Backend\Modules\MediaLibrary\Domain\MediaItem\Command\CreateMediaItemFromLocalStorageType;
 use Backend\Modules\MediaLibrary\Domain\MediaFolder\MediaFolder;
 use Backend\Modules\MediaLibrary\Component\UploadHandler;
+use Backend\Modules\MediaLibrary\Manager\ExtensionManager;
+use Backend\Modules\MediaLibrary\Manager\FileManager;
+use Backend\Modules\MediaLibrary\Manager\MimeTypeManager;
 use Common\Exception\AjaxExitException;
 use Common\Exception\RedirectException;
+use SimpleBus\Message\Bus\MessageBus;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\KernelInterface;
 
 /**
  * This AJAX-action is being used to upload new MediaItem items and save them into to the database.
  */
 class MediaItemUpload extends BackendBaseAJAXAction
 {
-    // override existing media
-    const OVERRIDE_EXISTING = false;
-
     /** @var Response */
     protected $response;
 
+    /** @var Request */
+    private $request;
+
+    /** @var LocalStorageProvider */
+    private $localStorageProvider;
+
+    /** @var FileManager */
+    private $fileManager;
+
+    /** @var ExtensionManager */
+    private $extensionManager;
+
+    /** @var MimeTypeManager */
+    private $mimeTypeManager;
+
+    /** @var MediaFolderRepository */
+    private $mediaFolderRepository;
+
+    /** @var MessageBus */
+    private $commandBus;
+
+    /** @var UploadHandler */
+    private $uploadHandler;
+
+    /** @var string */
+    private $uploadDirectory;
+
+    public function setKernel(KernelInterface $kernel = null): void
+    {
+        parent::setKernel($kernel);
+
+        if ($kernel === null) {
+            return;
+        }
+
+        $this->localStorageProvider = $this->get('media_library.storage.local');
+        $this->fileManager = $this->get('media_library.manager.file');
+        $this->extensionManager = $this->get('media_library.manager.extension');
+        $this->mimeTypeManager = $this->get('media_library.manager.mime_type');
+        $this->mediaFolderRepository = $this->get('media_library.repository.folder');
+        $this->commandBus = $this->get('command_bus');
+    }
+
     /**
      * PHP Server-Side Example for Fine Uploader (traditional endpoint handler).
      * Maintained by Widen Enterprises.
```

Comment:
We should probably remove this messagesince a lot of code changed :P

Reply:
removed in 3ef5b6992690eb67f0cfba70b71539223f9c8d59